### ISSUE TITLE
[Bug Report][3.6.8] VTreeview slow performance on expand

### ISSUE THREAD

**Issue by yuwu9145** [vuetifyjs/vuetify#19919]
### Environment
**Vuetify Version:** 3.6.8
**Vue Version:** 3.4.27
**Browsers:** Chrome 125.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
1. Load reproduction link
2. Expand a node

### Expected Behavior
Should be as fast as using VList: 
https://play.vuetifyjs.com/#eNp1k02P2jAQhv/KKKoEaMlCW/WShpVW26qnXlrtCZDWjQ24dWzLH5QK5b93kuDEhHBJ7JlnPt6xvT4n1hSLZ60fj54lWZI7VmpBHHvaSID8mBbEUCgEsXa1ScpTSrxTmwT+cuoOaPm4XG6Shm1owa2DY7pTBn2/gUv4EAEdku6N8hqORHiG4DMtuYwo5EIbmMwK5TJSOH4krsl7Bm2UtlBdhfTJOQZj3C8uKdINix077kRcLF9EeFx6EY0gMkd0XDSIndaOOfBZLbpeY82Yy/6wf4jxgTV0VUc8NpsBEGbUAJzG3kU/1qClmWs4jouxPUnc1EeJm4HA3BaG65ZiJ62MA8p2xAsH5zYRJY5kMJ3B6gmmFxuA0kxmsJ68WmbsZDsPdlLP16Knb3Q9+U4k2bOSSTeZw6SkPCVFobx0aYmFuBYsVd4JLlmUqQ78yZzjcm9DWKH2I2S/Koynw+IvhqHYkEELb+8U+8EIDdiO3+3pVeNEuny+3V0jX5hgPULb3W2/1eyyKFSpvWM0CzOH9g6FsXdWPPUF0hKf2TfmXg5cUMMkrIDTW67OW5NSUWaRWW8Hbry6MMXegKN3+Rl/Obz/VC8eHmY3yaBN9Ki9PfT3oHNiy9j/27szp1WK3+otGknHNHccsa8ngpeQQVFLaMTCnZhqNjBVg71hzhvZNnflisD7gxidwXJkBvfkB+U8an1M6ajGa3lRw2OiqnBtmj/C+aJ7vEk1l16I5rP9DxkApl0=

### Actual Behavior
Slow and frozen

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqVkt1ugzAMhV/FQrsA9YfulgHSNE17iFKpiLhqpBSiYFgllHefgTFSVC52E2P7+MgfybHzalOE71rv2wa9yIsJb1rlhGlWAsTtjgxiK/EbIsmtOsm8IWZeGodzl9Vx6IxyWhdGahps8K4rQyDwkjeKoOtrAEV10w2hiKYCwGAdgR9Aks7VXlrWBF9IH1ephMESEpBiIZpkZSWwZsHx5PYulQFfIYHk1uGNQwyvh/5jswkebWC02OumvvqLDu/IC59fOinsjk973i4FJEkhaz7vOf8PhKLfeUCDZwM2cHPrJgapMeW4zVx3JOvA/8Bdh+1R5WLdJd0q14DNVxW59+bL4EHmwjtcz8Dt7+AURU759FT8zo6+7BGHfy/Ps9uyUWo4Tj9cQ9D5)

**Comment by KaelWD** [comment]
The equivalent list example is just as slow https://play.vuetifyjs.com/#eNp1k02P2jAQhv/KKKoEaAmhrXpJA9JqW/XUS6s9AdK6sQG3jmP5g1Kh/PdOEpyYEC7EM/PMxzs2m0tkdJ48K7U4ORalUWZZoQSxbL2VANkpzommkAtizGobFeeYOFtuI/jLqT2i5+NyuY0atqEFN/ZqdXZ80KVTcCLCMcx4pgWXWOEU70uN9m/gEj4EZZpUPwZiRpQ2JbnlJ2KbjAsoXSoD1U1K349jMub94pIi3bDYz3Irgv7rLAnwsHUSrCBwB3TY1MuY1oE58Fktpz5jz5BL/7B/iPGB109VZywaYwD4tTUAp2E06TfttTSr9tdxdbY3iUZ9lWgMBGYm11y1FDurUlugbE+csHBpC1FiSQrTGazWML36AErFZAqbyath2kx2c+8n9X4NRvpBN5PvRJIDK5i0kzlMCspjkuelkzYusBFXgsWls4JLFlSqE38ya7k8GJ+Wl4cRsj/l2tFh8xfNUKyvoIQzD5r9YIR6bM8fzvSqcCNdPddat8gXJliP0Na6n7eaXQ95WShnGU39zqF9Q37tnRdvPUFaGgvfmH05ckE1k7ACTu+5um5NypIyg8xmNwjj04UpzgYco8vP+Mng/af68PQ0uysGbaGFcubYv4MuiCPj/G/vLpxWMf5Wb8FKOqZ544h9PRN8hAzyWkIjFh7kVLOBqxrYmlmnZTvcTSgAHy9idAfLkR08ku+V82D0MaWjGm/lBQOPiar8s2m+CGdJ9+eNqrl0QjQ/u/80UKZd

**Comment by KaelWD** [comment]
The equivalent list example has 200 list groups not 200 lists https://play.vuetifyjs.com/#eNp1Uk1vGjEQ/SujVSRAYYG2t+0SKWp77KVVToCEuzbEqde2/EGp0P73jL2YNSThwM7He/NmxrM6FdY080etZwfPiqqoHWu1II49rCVAfSgbYig0gli7XBftsSTeqXUB/zh1zxj5slisi4iNaMGtO3sXv9wb5TVUByI8Q8r2kbZc3p1eui0WOpQ7ZTD6AlzC56xarJC6QZgVylWkcfxAXGScQBulLXRXlEGWIxl5f7ikiI5Y1HPcidBFbAKZ9TyD59LzbBNZOEPnommMcUhMgU/COMFGzRxX/WX/EcZvoqmrwJhF5waQlhcBnObZ+bDwNEvceHqVc7B/UHTCi6JzM2BtG8N1j2JHrYwDynbECwenvhAljlQwnsDyAcbnGIDSTFawGj1ZZuxoM01xEvZrMTM0uhr9JJLsWcukG01h1FJekqZRXrqyRSGuBSuVd4JLllUKxN/MOS73NtEatX8HOViN8fRW/JthOGyqoIW3H4j9YoQm2I5/2NOTxo1c6vneu4Z8Z4INENp7b/vtJmejUa32jtEq7Rz6G0prv0QDVFoHUlFmYQmrzZDBQ4QxKgHHxOIrfmr4tAjG/f0kLwE9faa9fR4e9CyLPfBslvCLd1nB9seR4OGw2BrcnXi3vQJ2k8HrBtMw543sJVO0SwuIXwTX88sZFt1UeiHi3+YV08BbjQ==

Still faster though, treeview seems to be rerendering every node at least twice.

**Comment by yuwu9145** [comment]
Yeah, there are unnecessary renders

**Comment by yuwu9145** [comment]
Partially caused by https://github.com/vuejs/babel-plugin-jsx/issues/712

**Comment by uke5tar** [comment]
+1 on this issue. It is also a huge problem for me. I am currently trying to mitigate it by loading all items initially with an empty children array and then using the load-children prop to "fill" the children with the remaining items by resolving a fake Promise. It's a rather hacky workaround and not yet fully working for me as I need it but the only way to get a decent speed.

Any ideas on timeline for resolving this issue?

**Comment by TheInvoker** [comment]
+1

**Comment by yuwu9145** [comment]
Concluded two redundant renderings, on every open event:

- Treeview re-renders, caused by v-model:opened change
- Treeview item re-renders, caused by https://github.com/vuejs/babel-plugin-jsx/issues/712

**Comment by uke5tar** [comment]
Hi,
just want to let you know that even after upgrading to vuetify 3.6.9 the performance of the treeview is still quite slow with a big amount of items. I assume thats why you re-opened the issue @yuwu9145 ?
Any ideas on the timeline for fixing this issue?

Thank you for working on this btw :)

**Comment by yuwu9145** [comment]
> Hi,
> 
> just want to let you know that even after upgrading to vuetify 3.6.9 the performance of the treeview is still quite slow with a big amount of items. I assume thats why you re-opened the issue @yuwu9145 ?
> 
> Any ideas on the timeline for fixing this issue?
> 
> 
> 
> Thank you for working on this btw :) 

The remaining latency is caused by vuejs/babel-plugin-jsx#712

**Comment by jakeflorentine** [comment]
https://play.vuetifyjs.com/#eNq1Vm1v2jAQ/isnNKlB4y2sL2tEK7Huy75sX6rtQ1upxj6EJcfOHCeAEP99NhBiSHip1n4hOL7nufPdPec8LRqppt1hknTyDBtRY2AwTgQxeP8sAQZ522jEnOMUIm630rvnxur53LgfdMtdaz3oelC7TKnmiVnRCDRAJ1wwjRLuQFq2odZkHvR7vV6zM+ZCBE0HAt+s+NuJSRIEOREt4GzWhLt7WDhTAI0m07JYgd2OILSUPfjsTFvFe8ONwAheHxwjPGJq4NPCGixfNyZL91huQsBZorQBhmOSCVOwM2JIBMHKfVB6dLmI4KlYQxnMNqBtGF4oFzbjglNiuJIpRBc7NsXBd3hX3CvCfmvL8kAESkY0RECS5AKWOzwF4IsHmGgV43Hzy9L8D45So3RcD3jpUCXtGYIi4Ka37ZtWcnJVm5PvimYxSvOGhOwu1+TXe6cqHdgWN3w836c/6qLWTeHqpsLju7PKqnN10l1Ziq9lKbhkOLN1MGm1brug2xI0UsqkRpPkKPCl5m3Vct+q0jx1tQh7B4sR20mhORH99ytHuKu1969H6Gkvb4+MPKsgoSfAvE2JZufBPCHm7altAGWn8AfX0d89KuHwkIanUijC3jrUwuvysL+oUSN0Qy1h40NTKrwpAT9VjvFphKemx8yONdt8FjExsahMtrOzcFubhd+cofrPMdY/LJ0i+oqH09JZU4dlJr6RlFMQZK4yN3ghTi7r+2sD9RQwZDmRFBkYpBPJ/2Z4DoEnhqEQQEbWsbtdLJRxXQc9OXnWxJ5cfkijlYtF5Ye6oX/l3YpKjtGmjCJwh2QZdRezu/Vyfl5vFG+X6zvQflAMutsvoMayJTMhVj8v/wBzyliN

**Comment by lukinoe** [comment]
+1 

also facing big problems with performance with > 100 items

**Comment by Colorfulstan** [comment]
I managed to get the expand issue mitigated a bit by using mutliple VTreeviews with v-for. That way the performance issue does not scale that much and is only really noticeable in big branches of the tree-view.

However, I have another big Issue in terms of performance which is the initial rendering of a big treeview.
It blocks UI for a couple of seconds.

Is that the same Issue as already tackled here, or rather its own Ticket?

And any Tips on dealing with that? I'm sure we can come up with some ways when using the sub-components and basically re-build it with that with some kind of staggered addition to the DOM on the various trees or depth-levels, but maybe there is something planned for that already or something I don't see yet?

**Comment by robertzak** [comment]
Would it be possible to get an update on what the plans are for this issue? 

This is a major blocker for our migration to vue3 right now. I'd like to have an idea if I need to try to figure out an alternative to using `VTreeview`, or if I can modify the configuration in someway so it can support our tree of about 225 nodes, or if we have to stay on vuetify2 for the time being.

I saw a comment from @yuwu9145 on the upstream ticket (https://github.com/vuejs/babel-plugin-jsx/issues/712#issuecomment-2277145754) last week that setting the `stable` flag to `true`  "resolves our performance issues".

Is there anyway I can manually set that on the component or in the vuetify config? Reading the related docs on that ticket didn't make it clear to me how I could leverage that flag.

I see that 3.7 was released, but it seems like `VTreeview` remained in Labs, and now 3.8 is on hold/TBD due to funding issues? I assume that's because this one is a blocker for considering it a production ready component?

Thank you.

**Comment by PhilippeTh** [comment]
Hi,

I'm using the `VTreeview` with a looooot of nodes and, while I was coding to transition to Vue3 Vuetify3, I came across this very problem where the treeview is crashing my application. I thought I'd provide a minimum reproducible example with a single one of the trees I currently have in my application:

Here's the Vue2 Vuetify2 codesandbox (mdi font not working but example still works):
[Huge treeview CodeSandbox](https://codesandbox.io/p/sandbox/vuetify-2-forked-34793v?file=%2Fsrc%2FApp.vue)

Here's the Vue3 Vuetify3 version using play vuetify
Huge treeview Play vuetifyjs (link is too long I needed to put it in a txt file)
[LinkVuetifyPlayTooBig.txt](https://github.com/user-attachments/files/16638361/LinkVuetifyPlayTooBig.txt)

Something to note is also that adding the "activatable" flag on the newer version completely crashes it and you won't even be able to load it anymore (in my dev environment I see an infinite loop of `[Vue warn]: Failed to resolve directive: ripple ...`).

Here's my actual webpage running now where you can see the treeview working super smoothly with Vue2 Vuetify2 and with 4 of these huge trees instead of 1 like this example (2 in english and 2 in french):
https://eccc-msc.github.io/msc-animet

And the link to the github for the treeview I have that I'm absolutely unable to transition at the moment:
https://github.com/ECCC-MSC/msc-animet/blob/main/src/components/Layers/LayerTree.vue#L48

Hope this helps and hopefully this will unblock in the future.

Thank you!

**Comment by piliadis** [comment]
Hi there, 
Is there any update on this performance issue?
@robertzak did you managed to get it working?
Thank you!

**Comment by robertzak** [comment]
No. I've put it on hold. I couldn't figure out how to set the flag mentioned.

I also tried a few things to work around the slow performance, like only loading the root list and none of the child nodes, then dynamically adding them from a store when the node expands but other bugs with the tree component was preventing me from getting it working how it needs to.

At this point I'm just stuck either with the slow performance or simply removing the tree and coming up with an alternative way to show the data to our users

**Comment by piliadis** [comment]
Thanks @robertzak for the info. I'm in the same position, stuck using vuejs-tree.
Regards!

**Comment by lukinoe** [comment]
@pilladis

How we solved it was loading the nodes dynamically for each branch when the extend is clicked. In this way we could mitigate performance issues.

**Comment by robertzak** [comment]
That's my plan as well. However the load on expand wasn't working. It looks like they fixed that in `3.7.2` so I'm going to give it another try.

**Comment by klimekszymon** [comment]
> s loading the nodes dynamically for each branch whe

Could you provide with some code example, I would much appreciate.

I have to render tree with 200 parent nodes containing 5-10 childs each, without the performance optimisation the tree component is dramatically slow and impossible to interact with.

**Comment by johnleider** [comment]
> Still the same issue with big tree. Is there any plan for solving this issue?

The team is actively working on this.

**Comment by gilad7** [comment]
Maybe it will help someone as a workaround. I had the same issue when i put the treeview in an expandable item and tried to load the treeview in the onMounted phase. Until I wrapped the data loading code in setTimeout the item did not expand until the treeview was populated, even though I used v-if/v-else to show loading component while it is populating.

This did not work and did not expand the item only after a delay - (mapJsonObject can take some time to execute although I see the console.log pretty quick and the item expands after a few seconds):

```
onMounted(async () => {
  const parsedData = await mapJsonObject(props.data, 'metadata');
  console.log('Data parsed:', parsedData); 

  // Step 2: Assign the parsed data to the treeData array
  treeData.value = [parsedData];

  parsingData.value = false; // Update the parsingData flag after parsing is complete
});
```

This did work for me and immediately opened the item and showed the loading until treeview is populated then showed the treeview:

```
onMounted(() => {
  setTimeout(async () => {
    const parsedData = await mapJsonObject(props.data, 'metadata');
    console.log('Data parsed:', parsedData); // Logs the parsed data (for debugging or further processing)

    // Step 2: Assign the parsed data to the treeData array
    treeData.value = [parsedData];

    parsingData.value = false; // Update the parsingData flag after parsing is complete
  }, 50);
});

```

**Comment by Sphirye** [comment]
Any updates on this?  My browser dies for a few seconds when opening a single item in a 200 nodes treeview, which didnt happened in vuetify 2

https://github.com/user-attachments/assets/7d81ffff-411f-40b2-abdc-48e0bb1a80a1

**Comment by johnleider** [comment]
Has anyone been able to check performance against the latest version? The goal is still to get upstream fixed, but in the mean time, whatever micro optimizations that we can make to bring VTreeview closer to VList parity, we will continue to do so and reference this thread.

**Comment by robertzak** [comment]
I would have to rip out my workaround to do any testing. I may have some time next month to experiment with the latest, but not likely anytime sooner than that.

**Comment by Employee87** [comment]
I tried it in my application, but am seeing still large wait times for nodes to expand, like 5+ seconds.

**Comment by johnleider** [comment]
> I tried it in my application, but am seeing still large wait times for nodes to expand, like 5+ seconds.

If you're able to provide a playground I'll take a look.

**Comment by johnleider** [comment]
While slow in VTreeview, that's slow in VList as well. That many items without a virtual scroller isn't going to get much better by anything we can do (maybe someone smarter can figure it out).

**Comment by M2tias** [comment]
I've been checking the release notes and this issue for some weeks now hoping for any resolution to this problem. As you said the problem is not just VTreeview but the lack of virtual scrolling. This is pretty evident when you have checkboxes enabled and it seems to take a long time to redraw after a selection.

I have two levels of hierarchy in my treeview and as far as I know I can't implement virtual scrolling on top of the VTreeview component. The way VTreeview is built I don't see how it can be virtual scrolled easily because it has nested VLists inside. To be fair I don't see why the treeview couldn't be implement with just one VList. Just track the toggling/checkboxes separately and render each item of the list based on the depth in the tree. That way the virtual scroller could be implemented and the treeview would be usable with larger datasets.

**Comment by alextaymx** [comment]
@johnleider Thanks for the insight! Yeah, rendering that many items without virtualization will naturally be slow. Do you know if there's any workaround to enable virtual rendering or an approach to load children only when the parent node is expanded? That way, we could potentially reduce the initial render load and improve performance.

**Comment by johnleider** [comment]
We're at the edge of my knowledge on this subject. I'm certain there are things that can be done to make this better in it's current state, but I haven't been able to figure it out.

**Comment by DeclanMurphy** [comment]
After spending a few days in Chrome Performance tab, this is how I overcame a similar issue with the expand/collapse issue on **Vuetify v2**. I'm not sure if this applies to v3 so please delete this comment if it doesn't help anyone.

But basically instead of vue calling all of the require lifecycle methods for each treeview node, this solution moves the workload to the browser's rendering of `display: none` instead. 

`genChildrenWrapper() {
            if (!this.children) return null // remove this.isOpen condition here
            const children = [this.children.map(c => this.genChild(c, this.disabled))]
            return this.$createElement(
                'div',
                {
                    staticClass: 'v-treeview-node__children',
                    class: {
                        'd-none': this.isOpen === false, // place it here instead
                    },
                },
                children,
            )
        },`

The issue with this solution is that in some cases the vdom or Vuetify treeview cache is possibly out of sync so please use with caution. Will update if I make any further progress.

**Comment by KaelWD** [comment]
v3 already does that (using v-show), the downside is you frontload all that work to when the component is first mounted.

I missed 3cbe7a43d37ac19a0237824f660f48644bf9c908, this did improve it significantly and I'm not seeing the entire list re-render on open now.

**Comment by Employee87** [comment]
now that the stable flag fix has been merged, when will it come into vuetify?

https://github.com/vuejs/babel-plugin-jsx/issues/712

**Comment by johnleider** [comment]
> now that the stable flag fix has been merged, when will it come into vuetify?
> 
> [vue-macros/vue-macros#942](https://github.com/vue-macros/vue-macros/pull/942) [vuejs/babel-plugin-jsx#712](https://github.com/vuejs/babel-plugin-jsx/issues/712)

Whenever they release a new version.

**Comment by codetakki** [comment]
Would love to see this released soon, is there any ETA?

**Comment by nowrap** [comment]
Are there any news on this topic?

**Comment by KaelWD** [comment]
Replace vuetify in your package.json with `"vuetify": "npm:@vuetify/nightly@3.9.0-pr-21528.133e094"` and run npm install.
Don't use this in production yet, it might have introduced reactivity bugs in other components so I'd like some beta testers to post their results in #21528

**Comment by FAKER-A** [comment]
Hello, I see that the current data is fully rendered. Have you considered using virtual scrolling? @KaelWD

**Comment by MattTheCuber** [comment]
Have we made any progress on this issue? We are delivering our product in a few months and would love to know if we need to switch to a different component for our tree.

**Comment by KaelWD** [comment]
`items-registration="props"` from #22255 doesn't seem make a measurable difference in the original playground.

**Comment by J-Sek** [comment]
I vaguely remember choosing not to close this issue because I did not see the improvement I expected.

However with those demos: VTreeview [[1]](https://play.vuetifyjs.com/#eNqVU8tugzAQ/JWV1QMovKJWPVASqaqqfkQSKS5sGqsELHtJIyH+vQaHxkHJoRfMzs6OZ7BZtUyrPH6VMjo2yFKWER5kyQmX6wogO4akEI8Cf/oSQJi2DhV+CU2Kk6irxZpJVUu9ZpZRS6zCugrzUuTfFkqHKUMcVktcZvFF2+yVxc7GptS5EpIGE3iStSIocMebkqC1onl9kA1hkY7A2VwKng+L5QXtqZUm+EB624uyUFjBAkQxIY20qi5QG8Jq4/Z2tQKvRAJhWsmLWTKYJ/3LbOZfy4CViGSj996kYzwaw9uHVhRdaJ7dNpgSSFCJhvN+4uZ7IOS95yEa3BrofLfu3EIhNaqybi64Q7kf+B9x74fto4qJ3Wm6u7mG2OaoUvfcPOFf0dzwTq5bwbvz4LgWnPh4Vby2s7pGI4v/bh7rAvYYPUdJyEu559ET6+v53ACfSDxKWLDjpcbg6hey2OYXJETyXw==), VList [[2]](https://play.vuetifyjs.com/#eNplk9Fu4jAQRX9lZO0DLHFI0u1LFirtw2o/olRqNh6KUYgt26FIUf69Y7sB00pItmfO3DszKM8js6Zd/9E6Pw/IarZxeNJd4/Bp1wNszrxtjIC2a6zd7tjpwpvBqR2DdyncgSIPRbFjgQ10J62DWpKIx8NJcDi5wTfKmsZJ1VNSG6Up+VlL1bMznLntlKsP2Ag0RI4QWJgS+urGvTiV/Je9uKlSBWrsBZdtMDsJyVv1Ton1zXB9N6sPRMk4Oj387PS4A+lpWyN1pPCilXEgcN8MnYMxCrXqpAeHop4DEFdQw2IJ26db1KM9baxXAi1s4fnlltkrA4sOHUhKFL/p2EBV+MtqtUwlUhFCR5BkLDNw0nVYw+s/owYNP0Y5vWbQHmQnDPY1edFCE5Gr3zH6Hcmv9H7Hb34QvPJZK9eDPSy+EDSybwN+epECVnDMvgJzg38vDa0Xw458n0kNtXxfNS3TNw2QPsMaYzf+mqDJpAbdYPrIztHp0yWeBG/W1z+ZTRl7yB/z6hfzl7KgHxd4zquieuRlycuKZfums5jdfUsx9vIB5E/7QQ==) - both seem very fast.
Could you share a playground that is still slow?

**Comment by johnleider** [comment]
> I vaguely remember choosing not to close this issue because I did not see the improvement I expected.
> 
> However with those demos: VTreeview [[1]](https://play.vuetifyjs.com/#eNplk9Fu4jAQRX9lZO0DLHFI0u1LFirtw2o/olRqNh6KUYgt26FIUf69Y7sB00pItmfO3DszKM8js6Zd/9E6Pw/IarZxeNJd4/Bp1wNszrxtjIC2a6zd7tjpwpvBqR2DdyncgSIPRbFjgQ10J62DWpKIx8NJcDi5wTfKmsZJ1VNSG6Up+VlL1bMznLntlKsP2Ag0RI4QWJgS+urGvTiV/Je9uKlSBWrsBZdtMDsJyVv1Ton1zXB9N6sPRMk4Oj387PS4A+lpWyN1pPCilXEgcN8MnYMxCrXqpAeHop4DEFdQw2IJ26db1KM9baxXAi1s4fnlltkrA4sOHUhKFL/p2EBV+MtqtUwlUhFCR5BkLDNw0nVYw+s/owYNP0Y5vWbQHmQnDPY1edFCE5Gr3zH6Hcmv9H7Hb34QvPJZK9eDPSy+EDSybwN+epECVnDMvgJzg38vDa0Xw458n0kNtXxfNS3TNw2QPsMaYzf+mqDJpAbdYPrIztHp0yWeBG/W1z+ZTRl7yB/z6hfzl7KgHxd4zquieuRlycuKZfums5jdfUsx9vIB5E/7QQ==), VList [[2]](https://play.vuetifyjs.com/#eNplk9Fu4jAQRX9lZO0DLHFI0u1LFirtw2o/olRqNh6KUYgt26FIUf69Y7sB00pItmfO3DszKM8js6Zd/9E6Pw/IarZxeNJd4/Bp1wNszrxtjIC2a6zd7tjpwpvBqR2DdyncgSIPRbFjgQ10J62DWpKIx8NJcDi5wTfKmsZJ1VNSG6Up+VlL1bMznLntlKsP2Ag0RI4QWJgS+urGvTiV/Je9uKlSBWrsBZdtMDsJyVv1Ton1zXB9N6sPRMk4Oj387PS4A+lpWyN1pPCilXEgcN8MnYMxCrXqpAeHop4DEFdQw2IJ26db1KM9baxXAi1s4fnlltkrA4sOHUhKFL/p2EBV+MtqtUwlUhFCR5BkLDNw0nVYw+s/owYNP0Y5vWbQHmQnDPY1edFCE5Gr3zH6Hcmv9H7Hb34QvPJZK9eDPSy+EDSybwN+epECVnDMvgJzg38vDa0Xw458n0kNtXxfNS3TNw2QPsMaYzf+mqDJpAbdYPrIztHp0yWeBG/W1z+ZTRl7yB/z6hfzl7KgHxd4zquieuRlycuKZfums5jdfUsx9vIB5E/7QQ==) - both seem very fast. Could you share a playground that is still slow?

It's extremely noticeable on initial render time. A stark improvement from my point of view.

**Comment by PhilippeTh** [comment]
Well if you're looking for a playground with a huge tree there's always my old playground, that, as I'm trying it right now, still seems to be insanely slow: https://github.com/vuetifyjs/vuetify/issues/19919#issuecomment-2293652001

However the web page that is linked is no longer using Vue2, it's now using Vue3 except the tree nodes are a custom implementation and don't use v-treeview.

**Comment by J-Sek** [comment]
Did you apply `items-registration="props"` on it? The whole point of the improvement is to avoid rendering collapsed nodes. The initial playground won't ever be fast in its original state. + it requires 3.11.0 (currently beta)

**Comment by PhilippeTh** [comment]
@J-Sek Ahhhh Yup `items-registration="props"` wasn't doing it but after changing to 3.11 beta it does seem to work from my 16k nodes playground. Lovely!

**Comment by MaevskiyE** [comment]
Hi @KaelWD, @J-Sek  . I faced with some kind of related problem with performance. Please try to check items in the tree
Should I create new ticket? Thanks
Repro: [https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNrFWm1v2zYQ/iucOsAyZsuR5CSNkXQthm4rsA9DU+xLVbSKRdtq9eJRtFMj8H/fHSnJMm3RZmc4AQJb5N3x+PBeePJ9fLLidJ4z3k/DufO1yDNrZD0FGSFBOVEE1oiIERxbLig+B9aM83kxGgzGUQZsEU3iJXMyygfZPB28BrIBW2Q8Tmk/ytPXvnPpeJeDKC54c9yhRdp/YPljQRlICaxeY50BDC4p6zOaRZRRduy6CtvW2sqcdn0eT1ZHrFlSwjKu61zJZcqxfhI+FLiGkI2i10G2tnpWEmffCgXtcSGQ/lhp8P8WFdJQ1Ce5YsHGgzfzuQNEsOQtp+k8CTl9hSS3y34xo5STh5wBLITlC4AnEnNidpxnPIwzmJoki7iaEFOAXf1Y0iYE/ou7wHK9wCJpBN+uAqtBJeg4o3QZ08fmMCHLfprDVkcFTeiYU+StvsrtbP5GMWwClxGf6qzk6hecwS6nKyDjYHffVDLk7fOYJxQosjAF495DsAyThZZgPIuTCIxKqMuW8ZjuKMQoX7Csnz98Bb326Ro+JLQ5/up2sEGpiTEMA8CbQ4DnzSnI2fK45OnCiDheeLodNA4eHosxi+ccFOCL+dYBSdcnT6D2pEeKWZgk+eN7OiFrMmF5SjpgR50mA6xZoCB5VuQOGe2Pn7q7NOK4KgI5/UTiaETciwu3RxBj9Ll3iOuFCy5JPn/mqznFCTIKrHsJcGCtS2etuT2VG+yvlZsgeymgUrKU45eCiSLOr+MD2S/1D/CceYOosoXaqzfqens26/dLOfp9C813pO1sviHtIA470nyNNITBTNpQI21oLO1SI+3SWNqVRhqELUNp1xpp18bSXmqkvTSWdqORdmMqzb1ol+ZeGEvT+IJr7AuuxhcwJxlK0/iCa+wLrsYXXGNfcDW+4Br7gqvxBdfYF1yNL7jGvuBqfME19gVX4wuusS94Gl/wjH3B0/iCZ+wLnsYXPGNf8DS+4Ol9oRIG91H8aGTfKu3ueMZBd9ifvi9b0jf6ww+m7zeMhSsHbz72E0loNuWzEbm+ACWI/blH4i65e0XsWhGpige6kF9I3MBRKvSl1KeG7uene87ibGrHQO92nXkY3fOQcRsOL7DAgrrrLw0pLZBUBOtueeuSECko7/j4Qcfej/J1C8ro2adFWQ8yJDYdyNfnAXkL4p1QdTA+7Yf4pgViDFCnhNg9aMm+HuSb84O8c//QXjpaMYaLx16MxaXjrMFiqIXYdZ8BYjV1HL477ce4pZ4Sl6eT2vHVIZAv9SBvktn5QFaznv4SqEG5Je2Ja+BZLflKD/KZ0t4WyGrS099mNSC3ZD1xnz0ryNd6kJ8h7e3c0MW1/Acwbkl74l5+Voxf6jF+hqy3U2mI8uLQDb7mVisLUU4cza2mA1E+HM2tVg6iXDiaW42SHkbJo7nVKtnDmHg0txo8PAweR3OrVbCHoeJobtWnPJ1P7XCrVa6HLnQst6/amm9ia75qa76Jrfmqrfkmtuartuab2Jqv2ppvYmu+amu+ia35qq35Jrbmq7bmm9iar9qab2JrvmprvomtDVVbGx5la9uJY9hylx+e+i6Pdb0+c9xoM4d3cY7MIV/cKBB5LRChe50SIqzKtRBBRamF6Cz1zl6IWqqV4amrFSz49BDp3x953rNBNGyBCAPlSa3o8hBE+rc/3vBcEKkItRRjw1MXY4cR0r+68c5Sje1D6KoFIUx4p0QI969HSP/mxbt6LoRaas3hqWvNw4FI/9rEO0uxCb8cwNPtQLYsQLNC2dSSQqeDwwtoaql7FsaMQocDdLso3Qo1QdkpU087Ve/Mhgi5q8myb2YzCWOiZQJGgkx2NYRAf7dZ2YZ/UBdGHZifxFOHMpazP8MsSqCN5o5QBBpZ84TKOZsCg2RZFNQuNarHUmjK4XbnBXzvwGC5e6GG3D6jab6k93yV0A/h1Aaav0H/uKAONG/YH5M8jH6HxhC72yP4/R0ubncR1CCbLLIxj/MMmjKaQojdlVYp98hh5I5E+XiR0ow7/y4oW92Ljg/QvvOi6j/iMwrdXQXKEH0nqC4Y0oTYP4GEbtkJg2PwCCYCfTP8bQI8IFOu/xu209hIDM1LW/qh5kI72bBkz6CLpNRRiiUZfSTlxm2bUcB3SXswiW03wrpL45Zbwkas5p7kAZba2B2cluoTQQr6JUDeaWyuMYnKwCx+NEbDKHq7BGl/QY8WhaYckAqb6KBOQrmm+F1aYRqCWGxA0tbqPuTRygGLgHY2GwWI+fVe1PDs6/MssdpG0+5ULWcToC2caZ5PExrOY+gmy9MBdJT9OgnTOFndvc8fcp6P4H7Sg/DagwDSgxDRg3dSPbj5ImC76wt7O1aB/c19aRQL1V5fO99RHfB9TlkcJhEt4mkW4wpOCuEA5kolSjcpTXNvnPhHCQabUBBk9LsglbZSRY3KzUtGGS2F0dd9khGdhIuEf5CDYEbTGe+UgW8wEIcET3hQ8NOEJRsUQVXZzWf1JmFS0O2evWqsVMLqPdVfRxXf+tN/bElmMQ==)

**Comment by J-Sek** [comment]
seems to work fine with `items-registration="props"`

**Comment by MaevskiyE** [comment]
> seems to work fine with `items-registration="props"`

Kindly ask team to document this prop :)

**Comment by J-Sek** [comment]
The props has description in API docs which is usually enough. But sure, we can also add a dedicated example, not sure if it will help, since some people did not seem to read this thread properly.

**Comment by semkeijsper** [comment]
`items-registration="props"` is just what I was looking for! thanks @J-Sek for implementing this, it is greatly appreciated.

**Comment by BotellaA** [comment]
`items-registration="props"` is perfect for the first rendering, but once you have rendered one node, could it be possible to cache those values and not wait again later when you reopen the same node?
